tests/iterator_helpers/reduce.js

```diff
@@ -0,0 +1,15 @@
+/* @flow */
+declare const iterator: Iterator<number>;
+declare const numberReducer: (number,  number) => number;
+declare const stringReducer: (string,  number) => string;
+declare const flexibleReducer: (string | number,  number) => string;
+
+// Basic usage
+(iterator.reduce(numberReducer): number); // OK
+(iterator.reduce(flexibleReducer): string | number); // OK
```

Comment:
This is the motivating example for the more complex type signature for reduce with no initial value.